tests/test_bonfire.py

```diff
@@ -347,3 +347,31 @@ def test_describe_wrong_ns(mocker):
         print(result.output)
     except FatalError:
         assert True
+
+
+def test_remove_defaults(mocker):
+    get_return_args = mocker.patch("bonfire.bonfire._get_return_args")
+
+    runner = CliRunner()
+    result = runner.invoke(bonfire.test, ["process", "some-app"])
+    assert result.exit_code == 0
+
+    kwargs = get_return_args.call_args.args[1]
+    assert kwargs["remove_resources"].select_all is True
+    assert kwargs["no_remove_resources"].select_all is False
+    assert kwargs["remove_dependencies"].select_all is False
+    assert kwargs["no_remove_dependencies"].select_all is True
```

Comment:
I like that you added a couple of tests with the feature, so kudos to that :+1: 

But - I think this is testing an internal implementation detail (like, check if some values are set to true). 

If the implementation changes this test is in need to change too, besides, the output could still have the resources stripped in case there's another bug and this test would still pass right ?

I'd say we need a test that processes a template and that asserts on the processed output. 

Again, thanks for adding some tests along the fix!!

Reply:
We are testing this in 2 parts, and I think when you combine the 2 parts you end up with the more "end to end" type test you are looking for. This set of tests covers that the options for `remove_resourcces` / `no_remove_resources` get set as expected when the CLI options are provided... then we cover if the `_should_remove` function behaves as expected and we test a wide range of combinations of `remove_resources` / `no_remove_resources` here: https://github.com/RedHatInsights/bonfire/blob/master/tests/test_processor.py#L405-L590